assets/scripts/lobby/buttonEventListeners2.js

```diff
@@ -224,10 +224,10 @@ $('#createNewRoomButton').on('click', (data) => {
   }
 });
 
-$('#joinQueueButton').on('click', () => {
-  socket.emit('join-queue');
+$('#QueueButton').on('click', () => {
+  if ($('#QueueButton').text() === 'Join Queue') {
+    socket.emit('join-queue');
+  } else if ($('#QueueButton').text() === 'Leave Queue') {
+    socket.emit('leave-queue');
```

Comment:
We could also join the queue messages together. 
E.g. `queue-request`. 

Yeah that sounds better. Could you combine it please? And let's rename the reply `queue-reply`.